src/synapso_core/reranker/modernbert_reranker.py

```
@@ -1,35 +1,55 @@
-            output = generate(self.model, self.tokenizer, texts_to_embed)
+            texts_to_embed = [query_text] + [result[1] for result in results]
+            async with model_manager.acquire(
+                ModelNames.MODERNBERT_EMBEDDINGS_MODEL
+            ) as model_provider:
+                await model_provider.ensure_loaded()
+                output = generate(
+                    model_provider.model,
+                    model_provider.tokenizer,
+                    texts_to_embed,
+                )
         except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)